README.md

```diff
@@ -9,76 +9,194 @@ Adrenaline
+ - **Redux:** Its super developer-friendly! I love an idea of middleware and higher-order stores. I'd like to keep using these. But if you want to use Relay you have to forget about this. It was true until Adrenaline :)
+ - **REST:** Do you have already have a stable API? Is it impossible or costly to migrate to GraphQL?  Well, don't fret because Adrenaline's adapter API helps you to implement a data fetching and caching strategy.
  - **Relay connections:** Most of the time I do not need connections. The problem is Relay forces me to use them.
- - **Relay mutations `getConfigs`:** As a developer you have no freedom to handle this cuz you can chose only predefined strategies like `RANGE_ADD`. In Adrenaline there is an ability to use more functional and extensible way to handle this.
+ - **Relay mutations `getConfigs`:** As a developer you have no freedom to handle this because you can chose only predefined strategies like `RANGE_ADD`. In Adrenaline there is an ability to use more functional and extensible way to handle this.
  - **Relay routes:** Imagine usage of Relay routes with react-router. If you want to move your view from one route to another you would have to fix it in two places: in RR routes and Relay routes. Here I found react-redux idea with smart component much better.
 
 ## Installation
 
 `npm install --save adrenaline`
 
-Adrenaline requires **React 0.13 or later.**
+Note: Adrenaline requires **React 0.14 or later.** Containers make use of 0.14's parent based context for propagating state changes.
+
+Adrenaline's GraphQL strategy  uses `fetch` under the hood so you need to install polyfill by yourself.
+
+## Concepts
+* Queries and Args - Containers define their data requirements with respect to
+  * `args(props): Object`) A transformation of props that affect changes in your container's query requirements. Args are optional, because queries can be computed directly from props. But, args
+  will be spread as props along the same lifecycle as the data resolved for queries. In this way, you
+  can be sure that certain props (like id's or filters) that are tightly coupled to queries are updated
+  when the associated data is loaded.
+  * `queries(args): Object`) A function that returns a map of queries (the schema for which should be understood by your adapter strategy for fetching data) keyed by prop names.
+* Adapter - The Adrenaline adapter describes the strategy for fetching and resolving data from
+  both your API and cache. The adapter will dispatch actions when necessary to cache data with
+  a paired reducer. When implementing your own reducer, it's recommended to keep the reducer
+  close and in mind due to the tight relationship between data resolution and cache actions.
+* Container - A container represents a node in your redux tree that describes and resolves
```

Comment:
:+1: yes. I wrote most of this in my post soccer game sleeplessness. There are probably other minor mistakes. I'll push some updates as I refine it.